CodeEditModules/Modules/Git/src/Client/Live.swift

```diff
@@ -59,6 +59,33 @@ public extension GitClient {
             }
         }
 
+        /// Displays paths that have differences between the index file and the current HEAD commit,
+        /// paths that have differences between the working tree and the index file, and paths in the working tree
+        func getChangedFiles() throws -> [ChangedFiles] {
+            let output = try shellClient.run(
+                "cd \(directoryURL.relativePath.escapedWhiteSpaces());git status -s --porcelain -u"
+            )
+            if output.contains("fatal: not a git repository") {
+                throw GitClientError.notGitRepository
+            }
+            return try output
+                .split(whereSeparator: \.isNewline)
+                .compactMap { line -> ChangedFiles in
+                    let paramData = line.trimmingCharacters(in: .whitespacesAndNewlines)
+                    let parameters = paramData.components(separatedBy: " ")
+                    guard let url = URL(string: parameters[safe: 1] ?? "") else {
```

Comment:
since we are guarding for the url we shouldn't handle the optional with an empty string. we can return an error or nil if we can't find what we are looking for on the array